mantis-control-plane/mantis-control-plane-client/src/main/java/io/mantisrx/server/master/client/HighAvailabilityServicesUtil.java

```diff
@@ -121,99 +123,103 @@ protected void startUp() throws Exception {
     protected void shutDown() throws Exception {
     }
   }
+    private static class HighAvailabilityServicesImpl extends AbstractIdleService implements
+        HighAvailabilityServices {
+
+        private final MasterMonitor masterMonitor;
+        private final Counter resourceLeaderChangeCounter;
+        private final Counter resourceLeaderAlreadyRegisteredCounter;
+        private final AtomicInteger rmConnections = new AtomicInteger(0);
+        private final CoreConfiguration configuration;
+
+        public HighAvailabilityServicesImpl(CoreConfiguration configuration) {
+            this.configuration = configuration;
+            if(configuration.getLeaderMonitorFactory() instanceof LocalLeaderFactory) {
+                log.warn("using default non-local Zookeeper leader monitoring you should set: "+
+                    "mantis.leader.monitor.factory=io.mantisrx.server.core.master.ZookeeperLeaderMonitorFactory");
```

Comment:
I am not sure I follow the logic here: if it's a local leader factory, then use ZookeeperLeaderMonitorFactory? Shouldn't local leader refer to a single node local setup?